documentation/proposals/Proposal - Generic Math.md

```diff
@@ -0,0 +1,1189 @@
+
+    ///<summary>
+    ///The Y component of the Vector
+    ///</summary>
+    [DataMember]
+    public TScalar Y { get; }
+
+    public static Vector2F<TScalar> UnitX { get; }
+    public static Vector2F<TScalar> UnitY { get; }
+
+    public static Vector2F<TScalar> Zero { get; }
+    public static Vector2F<TScalar> One { get; }
+
+    public TScalar this[int index] { get; }
+
+    public Vector2F(TScalar x, TScalar y);
+    public Vector2F(TScalar value);
+    public Vector2F(ReadOnlySpan<TScalar> span);
+
+    public Span<TScalar> AsSpan();
+
+    public static Vector2F<TScalar> operator +(Vector2F<TScalar> left, Vector2F<TScalar> right);
+    public static Vector2F<TScalar> operator -(Vector2F<TScalar> left, Vector2F<TScalar> right);
+    public static Vector2F<TScalar> operator *(Vector2F<TScalar> left, Vector2F<TScalar> right);
+    public static Vector2F<TScalar> operator /(Vector2F<TScalar> left, Vector2F<TScalar> right);
+    public static Vector2F<TScalar> operator %(Vector2F<TScalar> left, Vector2F<TScalar> right);
+
+    public static Vector2F<TScalar> operator *(Vector2F<TScalar> left, TScalar right);
+    public static Vector2F<TScalar> operator /(Vector2F<TScalar> left, TScalar right);
+    public static Vector2F<TScalar> operator %(Vector2F<TScalar> left, TScalar right);
+
+    public static Vector2F<TScalar> operator *(TScalar left, Vector2F<TScalar> right);
+
+    public static Vector2F<TScalar> operator +(Vector2F<TScalar> value);
+    public static Vector2F<TScalar> operator -(Vector2F<TScalar> value);
+
+    public TScalar LengthSquared { get; }
+    public TScalar Length();
+
+    public Vector2F<TScalar> Dot(Vector2F<TScalar> other);
+    public TScalar DistanceSquared(Vector2F<TScalar> other);
+    public TScalar Distance(Vector2F<TScalar> other);
+    public Vector2F<TScalar> Abs();
+    public void CopyTo(TScalar[] array);
+    public void CopyTo(TScalar[] array, int index);
+
+    public static Vector2F<TScalar> Dot (Vector2F<TScalar> left, Vector2F<TScalar> right);
+    public static Vector2F<TScalar> DistanceSquared (Vector2F<TScalar> left, Vector2F<TScalar> right);
+    public static Vector2F<TScalar> Distance (Vector2F<TScalar> left, Vector2F<TScalar> right);
+    public static Vector2F<TScalar> Clamp (Vector2F<TScalar> value, Vector2F<TScalar> min, Vector2F<TScalar> max);
+    public static Vector2F<TScalar> Min (Vector2F<TScalar> value1, Vector2F<TScalar> value2);
+    public static Vector2F<TScalar> Max (Vector2F<TScalar> value1, Vector2F<TScalar> value2);
+    public static Vector2F<TScalar> Abs (Vector2F<TScalar> value);
+}
+```
+At the end of this proposal is the definition for the new interfaces used in the these classes.
+
+# Matrix Types
+
+This proposal includes the following matrix types:
+- Matrix2x2F
+- Matrix2x2I
+- Matrix2x3F
+- Matrix2x3I
+- Matrix2x4F
+- Matrix2x4I
+- Matrix3x2F
+- Matrix3x2I
+- Matrix3x3F
+- Matrix3x3I
+- Matrix3x4F
+- Matrix3x4I
+- Matrix4x2F
+- Matrix4x2I
```

Comment:
Same question here. What is `3x4` used for? `4x3` is definitely common and is `4x Vector3` with the final column being implied `0, 0, 0, 1`.

I'm not familiar with any common use cases for a `4x2`, however.

Reply:
4x2, No clue.
3x4 may just be for transpose.

Reply:
as above the additional types are needed for opengl bindings